cacheops/redis.py

```diff
@@ -29,24 +30,48 @@ def handle_connection_failure(call):
 
 client_class_name = getattr(settings, 'CACHEOPS_CLIENT_CLASS', None)
 client_class = import_string(client_class_name) if client_class_name else redis.StrictRedis
+redis_replicas = []
+
+
+def ip(url):
+    return socket.gethostbyname(urlparse(url).hostname)
+
+
+def set_redis_replicas():
+    # the conf could be a list or string
+    # list would look like: ["redis://cache-001:6379/1", "redis://cache-002:6379/2"]
+    # string would be: "redis://cache-001:6379/1,redis://cache-002:6379/2"
+    global redis_replicas
+    if isinstance(settings.REDIS_REPLICAS, six.string_types):
+        temp = settings.REDIS_REPLICAS.split(',')
+    else:
+        temp = list(settings.REDIS_REPLICAS)
+    master_url = settings.REDIS_MASTER
+    master_ip = ip(master_url)
+    temp = [r for r in temp if ip(r) != master_ip]
+    temp = [r for r in temp for _ in range(settings.REDIS_REPLICA_WEIGHT)]
+    temp = map(redis.StrictRedis.from_url, temp)
```

Comment:
`from_url` returns a client object as per `https://redis-py.readthedocs.io/en/latest/#redis.StrictRedis.from_url`. Is this what we want here?
Another concern: cacheops uses db=1, while from_url will give db=0 by default.

Reply:
Yes, `redis_replicas` is a list of Redis client objects – see https://github.com/HearstCorp/django-cacheops/blob/PDT-618-weighted-replica-round-robin/cacheops/redis.py#L107 where we use one to `.get()` from the cache:
```py
    class ReplicaProxyRedis(CacheopsRedis):
        """Proxy `get` calls to redis replica."""
        def get(self, *args, **kwargs):
            try:
                redis_replica = random.choice(redis_replicas)
                return redis_replica.get(*args, **kwargs)
            except redis.ConnectionError:
                return super(ReplicaProxyRedis, self).get(*args, **kwargs)
```

Reply:
But you know what I am doing that's dumb – I'm adding the extra URLs then making client objects for all of them, instead of making client objects first and duplicating the references! Will add that...